src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java

```diff
@@ -166,7 +150,18 @@ public void start(Map<String, String> props) {
 
             url = config.getString(InterfaceConst.URL);
             String userName = config.getString(InterfaceConst.USER_NAME);
-            String password = config.getPassword(InterfaceConst.PASSWORD).value();
+            if (!Objects.requireNonNull(userName).isEmpty()) {
+                String password;
+                //Password pass = config.getPassword(InterfaceConst.PASSWORD);
+                if (!Objects.requireNonNull(config.getPassword(InterfaceConst.PASSWORD)).value().isEmpty()) {
+                    password = config.getPassword(InterfaceConst.PASSWORD).value();
+                } else {
+                    throw new ConfigException("Password required in config.");
+                }
+                service = JavaCloudantUtil.getClientInstance(url, userName, password);
+            }  else {
+                throw new ConfigException("Username and password required in config.");
+            }
```

Comment:
Similar to the other validation piece - I think we should drop this to keep the migration "simple" and add it in the "Kafka way" later.